### PR TITLE
Sequelize table constructors

### PR DESCRIPTION
**Pull request by bdalberson**

### CODE REVIEW

File: models/Favorite.js
Comment (LiubovSobolevskaya) [inline models/Favorite.js:1]: What are lines 14-19 vs 21-28? SpoonID is only for recipes! use recipe_id instead! modelName: "comment"? Is it a good name for a table named Favorite?   
```
 SpoonID references: {
        **model: "post",**
        key: "id",
  },
```
      
We do not even have such a model!

File: models/Recipe.js
Comment (LiubovSobolevskaya) [inline models/Recipe.js:1]: lines 21-23 
   validate: {
        len: [1],
      }, 

Does Integer a length?

line 25 Use snake case. Spoon_id does not reference any other table.

File: models/index.js
Comment (LiubovSobolevskaya) [inline models/index.js:1]: We decided to get rid of comments.

**Review by LiubovSobolevskaya — CHANGES_REQUESTED** [review]

File: models/index.js
Comment (bdalberson) [reply models/index.js:1]: Believe I have resolved this.

File: models/Favorite.js
Comment (bdalberson) [reply models/Favorite.js:1]: Believe I have resolved this.

**Comment by bdalberson** [comment]
Believe I addressed the comments, Lets me know if I missed any

File: models/Favorite.js [hunk 1]
```diff
@@ -27,11 +26,11 @@ Comment.init(
...
       },
     },
-    Recipe_id: {
+    recipe_id: { //foreign key
       type: DataTypes.INTEGER,
-      allowNull: false,
+      allowNull: true,
       references: {
-        model: "recipe",
+        model: "Recipe",
```
Comment (LiubovSobolevskaya) [inline models/Favorite.js:33]: Recipe should not be capitalized. it is a model name from recipe.js line 39

File: models/Recipe.js [hunk 2]
```diff
@@ -11,19 +11,19 @@ Recipe.init(
...
     },
-    content: {
-      type: DataTypes.TEXT,
-      allowNull: false,
+    calories: {
+      type: DataTypes.INTEGER,
+      allowNull: true,
       validate: {
-        len: [1],
       },
```
Comment (LiubovSobolevskaya) [inline models/Recipe.js:22]: what is the purpose of this validate?

File: models/Recipe.js [hunk 3]
```diff
@@ -11,19 +11,19 @@ Recipe.init(
...
     },
-    user_id: {
+    user_id: { //foreign key
```
Comment (LiubovSobolevskaya) [inline models/Recipe.js:24]: Recepe does not reference users in any way here. Only through favorites in index.js

Comment (LiubovSobolevskaya) [reply models/Recipe.js:24]: where is spoon_id?

File: models/index.js [hunk 4]
```diff
@@ -12,21 +12,5 @@ User.hasMany(Recipe, {
...
-  Recipe.hasMany(Comment, {
-    foreignKey: "recipe_id", //foreign key relationship
-  });
-  
-  User.hasMany(Comment, {
-    foreignKey: "user_id", // foreign key relationship
-  });
- 
-  module.exports = { User, Recipe, Comment };
+  module.exports = { User, Recipe, Favorite };
```
Comment (LiubovSobolevskaya) [inline models/index.js:16]: no onDelete property. Please reference https://ucb.bootcampcontent.com/UCB-Coding-Bootcamp/UCB-VIRT-FSF-FT-03-2023-U-LOLC/-/blob/main/course-content/13-ORM/activities/28-Stu_Mini-Project/Main/models/index.js to do a many-to-many relationships.

**Review by LiubovSobolevskaya — CHANGES_REQUESTED** [review]

File: models/Favorite.js (hunk 1 above)
Comment (bdalberson) [reply models/Favorite.js:33]: removing

File: models/Recipe.js (hunk 2 above)
Comment (bdalberson) [reply models/Recipe.js:22]: gonna remove this

File: models/Recipe.js (hunk 3 above)
Comment (bdalberson) [reply models/Recipe.js:24]: adding that.

File: models/index.js (hunk 4 above)
Comment (bdalberson) [reply models/index.js:16]: I cant find the on delete in that reference.

File: models/Favorite.js [hunk 5]
```diff
@@ -1,23 +1,19 @@
...
+Favorite.init(
   {
     id: {
       type: DataTypes.INTEGER,
       allowNull: false,
       primaryKey: true,
       autoIncrement: true,
     },
-    comment_text: {
+    user_id: {
```
Comment (LiubovSobolevskaya) [inline models/Favorite.js:14]: why user_id twice?

File: models/Favorite.js [hunk 6]
```diff
@@ -27,22 +23,23 @@ Comment.init(
...
+    recipe_id: { //foreign key
       type: DataTypes.INTEGER,
-      allowNull: false,
-      references: {
-        model: "recipe",
-        key: "id",
-      },
+      allowNull: true,
     },
+    spoon_id:{
```
Comment (LiubovSobolevskaya) [inline models/Favorite.js:30]: not needed

Comment (LiubovSobolevskaya) [inline models/Favorite.js:30]: not needed. Add spoon_id instead

File: models/index.js (hunk 4 above)
Comment (LiubovSobolevskaya) [reply models/index.js:16]: OnDelete: https://ucb.bootcampcontent.com/UCB-Coding-Bootcamp/UCB-VIRT-FSF-FT-03-2023-U-LOLC/-/blob/main/course-content/13-ORM/activities/21-Ins_One-to-One/models/index.js


Use the previous reference to make many-to-many. Many recipes are saved by many users. Many users can save many recepies.

**Review by LiubovSobolevskaya — CHANGES_REQUESTED** [review]

File: models/index.js (hunk 4 above)
Comment (bdalberson) [reply models/index.js:35]: Let me know if there's still issues.

File: models/Favorite.js (hunk 5 above)
Comment (bdalberson) [reply models/Favorite.js]: removing extra one

File: models/index.js (hunk 4 above)
Comment (LiubovSobolevskaya) [reply models/index.js:35]: There are still issues not addressed in favorite.js (spoon_id is not needed, recipe_id does not reference the corresponding table), index.js is not correct. Recipe misses spoon_id. I'll merge and fix it myself.

**Review by LiubovSobolevskaya — APPROVED** [review]

**Review by LiubovSobolevskaya — APPROVED** [review]

**Review by LiubovSobolevskaya — APPROVED** [review]

**Pull request merged**